### PR TITLE
Addressing Docker image version issue with hotfix deployments (SCP-4796)

### PR DESCRIPTION
**Pull request by bistline**
#### BACKGROUND
As a part of the normal hotfix process, one of the last steps before applying the fix to production is to run a test deployment of the fix against our staging host.  This process has been broken ever since #1552 was merged, as now the `development` tag is used exclusively on staging deployments.

#### CHANGES
This update fixes the issue of the incorrect `development` Docker image tag being pulled.  Now, the most recent production release will be used.  The `bin/deploy.sh` script has now been updated to allow for this by adding the `-f` option to specify `HOTFIX=true`.

This now means that in order to test a hotfix, we must first merge the associated PR in `master` and publish a new release.  This will trigger building the new Docker image tag, which will then be used in the test deployment to staging.  Our [release engineer playbook](https://docs.google.com/document/d/1rKcOQPtkbhZCbTqxTnaT-4E29VLzkJrm0kAoWjLx3QA/edit#heading=h.fojnq9y2hk31) has been updated to reflect the new procedure.

A [manual test](https://scp-jenkins.dsp-techops.broadinstitute.org/job/scp-deploy-hotfix-to-staging/129/console) of this on staging showed that the most recent tag was used instead of `development`.

```
1.41.0: Pulling from broad-singlecellportal-staging/single-cell-portal
Digest: sha256:148d192553608a860efde1d970d042b37cb7b854915a25d10aa131e9d514de70
Status: Downloaded newer image for gcr.io/broad-singlecellportal-staging/single-cell-portal:1.41.0
gcr.io/broad-singlecellportal-staging/single-cell-portal:1.41.0
### Stopping running container single_cell ###
single_cell
### Removing docker container single_cell ... ###
single_cell
### COMPLETED ###
### loading env secrets ###
### COMPLETED ###
### Booting single_cell with Docker image version 1.41.0 ###
### BOOTING PORTAL IN 'staging' MODE WITH DOCKER IMAGE VERSION 1.41.0
```
#### MANUAL TESTING
1. Sign onto the VPN and access our Jenkins host
2. Run the [scp-deploy-hotfix-to-staging](https://scp-jenkins.dsp-techops.broadinstitute.org/job/scp-deploy-hotfix-to-staging/) job
3. Confirm you see the above message in the console output
4. (Optional) Once completed, ssh into the staging host and verify that the correct Docker image was used:
```
sudo docker inspect single_cell | jq .[0].Config.Image
"gcr.io/broad-singlecellportal-staging/single-cell-portal:1.41.0"
```
5. Revert the staging host by running the [scp-deploy-development-to-staging](https://scp-jenkins.dsp-techops.broadinstitute.org/job/scp-deploy-development-to-staging/) job

### CODE REVIEW

File: bin/deploy.sh [hunk 1]
```diff
@@ -12,146 +12,149 @@ THIS_DIR="$(cd "$(dirname -- "$0")"; pwd)"
...
+  if [[ $ROLLBACK == "true" ]]; then
+    # checkout the requested tag (usually current release - 1)
+    echo "### ROLLING BACK DEPLOYMENT BY $TAG_OFFSET RELEASE TAG ###"
+    echo "### determining requested release rollback tag on $GIT_BRANCH ###"
+    ROLLBACK_TAG=$(extract_release_tag $TAG_OFFSET) || exit_with_error_message "could not get rollback tag in $GIT_BRANCH"
+    echo "### rolling back release to tag: $ROLLBACK_TAG on $GIT_BRANCH ###"
+    run_remote_command "git checkout tags/$ROLLBACK_TAG" || exit_with_error_message "could not checkout tags/$ROLLBACK_TAG on $GIT_BRANCH"
+    VERSION_TAG="$ROLLBACK_TAG"
     echo "### COMPLETED ###"
+  fi
```
Comment (eweitz) [inline bin/deploy.sh:144]: Per your standup comment, it seems the effective changes are limited to this section, with the other changes are less relevant indentation refinements.

File: bin/deploy.sh [hunk 2]
```diff
@@ -167,26 +170,27 @@ USAGE:
 -S VALUE    set the path to SSH_KEYFILE (private key for SSH auth, no default, not needing except for manual testing)
 -h VALUE    set the DESTINATION_HOST (remote GCP VM to SSH into, no default)
 -R          set ROLLBACK to true to revert release to last known good release
+-f          set HOTFIX to true to deploy the latest release tag to a non-production host
```
Comment (eweitz) [inline bin/deploy.sh:173]: Ah, and this change seems relevant too.

**Review by eweitz — APPROVED** [review]
Code and documentation updates look good!

For reviewers' convenience, I marked the changes that seem most relevant, i.e. non-indentation changes.

**Review by jlchang — APPROVED** [review]
Manual testing confirm [scp-deploy-hotfix-to-staging](https://scp-jenkins.dsp-techops.broadinstitute.org/job/scp-deploy-hotfix-to-staging/) works again. Yay!

**Pull request merged**